lib/firehose/long_poll_transport.coffee

```diff
@@ -108,7 +108,7 @@ class LongPollTransport extends Transport
 
     unless @_needToNotifyOfReconnect or @_stopRequestLoop
       @_needToNotifyOfReconnect = true
-      @config.disconnected()
+      @config.disconnected(jqXhr, status, error)
```

Comment:
Also, make sure we can determine if it was long poll or web socket in the handler. If the event doesn't include something we can use (somewhat clearly) then I'd suggest another parameter of just `long-poll` or similar